libshaderc/src/shaderc_test.cc

```diff
@@ -283,6 +289,78 @@ TEST_F(CompileStringWithOptionsTest, DisassemblyOption) {
   EXPECT_THAT(disassembly_text_cloned_options, HasSubstr("MemoryModel"));
 }
 
+TEST_F(CompileStringWithOptionsTest, ForcedVersionProfileCorrectStd) {
+  // Forces the version and profile to 450core, which fixes the missing
+  // #version.
+  shaderc_compile_options_set_forced_version_profile(options_.get(), 450,
+                                                     shaderc_profile_core);
+  ASSERT_NE(nullptr, compiler_.get_compiler_handle());
+  EXPECT_TRUE(CompilesToValidSpv(kCoreVertShaderWithoutVersion,
+                                 shaderc_glsl_vertex_shader, options_.get()));
+}
+
+TEST_F(CompileStringWithOptionsTest,
+       ForcedVersionProfileCorrectStdClonedOptions) {
+  // Forces the version and profile to 450core, which fixes the missing
+  // #version.
+  shaderc_compile_options_set_forced_version_profile(options_.get(), 450,
+                                                     shaderc_profile_core);
+  // This mode should be carried to any clone of the original options object.
+  compile_options_ptr cloned_options(
+      shaderc_compile_options_clone(options_.get()));
+  ASSERT_NE(nullptr, compiler_.get_compiler_handle());
+  EXPECT_TRUE(CompilesToValidSpv(kCoreVertShaderWithoutVersion,
+                                 shaderc_glsl_vertex_shader,
+                                 cloned_options.get()));
+}
+
+TEST_F(CompileStringWithOptionsTest, ForcedVersionProfileWrongStd) {
+  // Forces the version and profile to 310es, which is a wrong pair.
```

Comment:
Fix the comment, and maybe the test name.
It's not that 310es is a wrong pair.  It's that the module is invalid for that version of GLSL.